### PR TITLE
Option for decaying learning rate + util.py refactor 

### PR DESCRIPTION
**Pull request by deanrexines**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -78,16 +78,21 @@ def train(args):
         raise ValueError('save_model needs an explicit, existing parent directory: ' + args.save_model)
 
     records, n_examples = get_records(args.train_dir)
+
+    # set learning rate
+    learning_rate = args.learning_rate if not args.decay_lr else decaying_learning_rate(args.batch_size, n_examples, args.learning_rate)
```
Comment (brett-daley) [inline main.py]: I like how you did this here

File: util.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    return image_batch
+
+def rgb2gray(image_batch):
+    return tf.image.rgb_to_grayscale(image_batch)
+
+# optimization options
+def decaying_learning_rate(batch_size, decay_step, base = 0.01):
+    decay_rate = 0.95
+    staircase = True 
+    batch = tf.Variable(0, dtype=tf.float32)
```
Comment (brett-daley) [inline util.py]: Are you sure this works? I implemented a decaying learning rate once, and I remember you had to explicitly pass this variable (with `trainable=False`) into the GradientDescentOptimizer call

File: util.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    image_batch = tf.map_fn(lambda img: tf.image.random_flip_left_right(img), image_batch)
+    image_batch = tf.map_fn(lambda img: tf.image.random_flip_up_down(img), image_batch)
+
+    return image_batch
+
+def rgb2gray(image_batch):
+    return tf.image.rgb_to_grayscale(image_batch)
+
+# optimization options
+def decaying_learning_rate(batch_size, decay_step, base = 0.01):
```
Comment (brett-daley) [inline util.py]: Picky but I think we should use a better name than "base" for the initial learning rate

File: util.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    decay_rate = 0.95
+    staircase = True 
+    batch = tf.Variable(0, dtype=tf.float32)
+
+    learning_rate = tf.train.exponential_decay(
```
Comment (brett-daley) [inline util.py]: Also here I think we should explicitly have the kwargs because it's a little hard to understand without looking at the docs for `tf.train.exponential_decay`
